### ISSUE TITLE
Add some additional information about PN interactions in the event header

### ISSUE THREAD

**Issue by EinarElen** [LDMXAnalysis/ldmx-sw#1378]
We already track and store some information like 
```
  eventHeader.setFloatParameter("total_photonuclear_energy",
                                event_info->getPNEnergy());
  eventHeader.setFloatParameter("total_electronuclear_energy",
                                event_info->getENEnergy());
  eventHeader.setFloatParameter("db_material_z",
                                event_info->getDarkBremMaterialZ());
```

I think the following properties (at least) would be useful to be able to check and are currently not available at all

- Target nucleus (Z,A) for the PN interaction 
PN interactions behave really quite differently for different targets, so being able to understand what effect our materials have on the expected rates for different event types would be useful. Since we are wrapping Bertini, we can always access this. 
- The number of resample attempts that were used (if resampling was used, otherwise just 1)
Necessary for getting a better understanding of how to do EoT-estimates

**Comment by tomeichlersmith** [comment]
I'm switching the label here from `framework` to `simcore` since implementing this mainly relies on adding more information to the `UserEventInformation` object and then copying it into our `EventHeader` like is already done with the other `UserEventInformation` stuff. (i.e. in order to implement this, no change to Framework would be required).

**Comment by tvami** [comment]
@EinarElen are you actually working on this btw?

**Comment by EinarElen** [comment]
No, but it would be well suited together with the discussion we had about moving bertini into ldmx-sw, so i would link it to that
